src/main/java/uk/nhs/digital/nhsconnect/nhais/mesh/MongoScheduler.java

```diff
@@ -1,73 +1,49 @@
 package uk.nhs.digital.nhsconnect.nhais.mesh;
 
-import com.mongodb.client.result.UpdateResult;
+import java.time.LocalDateTime;
+
+import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
-import org.bson.Document;
+import uk.nhs.digital.nhsconnect.nhais.repository.SchedulerTimestampRepositoryExtensions;
+
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.data.mongodb.core.MongoOperations;
-import org.springframework.data.mongodb.core.query.Update;
 import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Component;
 
-import java.time.LocalDateTime;
-
-import static org.springframework.data.mongodb.core.query.Criteria.where;
-import static org.springframework.data.mongodb.core.query.Query.query;
-
 @Component
 @Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class MongoScheduler {
 
-    @Autowired
-    private MongoOperations mongoOperations;
+    private final MeshClient meshClient;
+    private final SchedulerTimestampRepositoryExtensions schedulerTimestampRepository;
```

Comment:
You can use the base `SchedulerTimestampRepository` as it also have what is in the extensions